coordinator/src/orderbook/routes.rs

```diff
@@ -19,57 +27,139 @@ use tokio::sync::broadcast::Sender;
 use tokio::sync::mpsc;
 use trade::Direction;
 
-#[derive(Serialize, Deserialize, Debug, Clone)]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
 pub struct Order {
     pub id: i32,
     #[serde(with = "rust_decimal::serde::float")]
     pub price: Decimal,
-    pub maker_id: String,
+    pub trader_id: String,
     pub taken: bool,
     pub direction: Direction,
     #[serde(with = "rust_decimal::serde::float")]
     pub quantity: Decimal,
+    pub order_type: OrderType,
 }
 
-pub async fn get_orders(State(state): State<Arc<AppState>>) -> impl IntoResponse {
-    let mut conn = state.pool.clone().get().unwrap();
-    let order = orderbook::db::orders::all(&mut conn).unwrap();
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub enum OrderType {
+    Market,
+    Limit,
+}
+
+pub async fn get_orders(State(state): State<Arc<AppState>>) -> Result<Json<Vec<Order>>, AppError> {
+    let mut conn = get_db_connection(&state)?;
+    let order = orderbook::db::orders::all(&mut conn)
+        .map_err(|e| AppError::InternalServerError(format!("Failed to load all orders: {e:#}")))?;
+
+    Ok(Json(order))
+}
 
-    Json(order)
+fn get_db_connection(
+    state: &Arc<AppState>,
+) -> Result<PooledConnection<ConnectionManager<PgConnection>>, AppError> {
+    state
+        .pool
+        .clone()
+        .get()
+        .map_err(|e| AppError::InternalServerError(format!("Failed to get db access: {e:#}")))
 }
 
 pub async fn get_order(
     Path(order_id): Path<i32>,
     State(state): State<Arc<AppState>>,
-) -> impl IntoResponse {
-    let pool = state.pool.clone();
-    let mut conn = pool.get().unwrap();
+) -> Result<Json<Order>, AppError> {
+    let mut conn = get_db_connection(&state)?;
     let order = orderbook::db::orders::get_with_id(&mut conn, order_id)
-        .unwrap()
-        .unwrap();
+        .map_err(|e| AppError::InternalServerError(format!("Failed to load order: {e:#}")))?
+        .context(format!("Order not found {order_id}"))
+        .map_err(|e| AppError::BadRequest(format!("{e:#}")))?;
 
-    Json(order)
+    Ok(Json(order))
 }
 
-#[derive(Deserialize, Serialize)]
+#[derive(Deserialize, Serialize, Clone)]
 pub struct NewOrder {
     pub price: Decimal,
     pub quantity: Decimal,
-    pub maker_id: String,
+    pub trader_id: String,
     pub direction: Direction,
+    pub order_type: OrderType,
+}
+
+pub struct MatchParams {
+    // todo: waiting for https://github.com/get10101/10101/pull/244 to land
```

Comment:
Please note that https://github.com/get10101/10101/pull/244 will not be merged; parts of the refactoring might eventually be moved to a separate PR, but the model and flow changes will not be incorporated for now.

The `TradeParams` are defined on `main` already. The output of matching is expected to be one `TradeParams` for each trader.
Note that https://github.com/get10101/10101/pull/228 adds the `Direction` to the `TradeParams` and changes the `expiry` to be a timestamp (`OffsetDateTime`) instead of a `Duration`.